CONN-1082

christophermay07 opened the pull request:
- Fixed issue where Hibernate was not persisting new entities when id was preset.
- Removed flush statements in favor of letting container handle it automatically.
- Whitespace fixes (tabs to spaces, trimmed trailing space).

christophermay07 committed:
CONN-1082: Allowing Hibernate to generate id (primary key)

msweaver commented on Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/impl/CertPolicyDaoImpl.java:
```diff
@@ -316,19 +312,19 @@ public CertPolicyGroup getPolicyGroupById(long id) throws ConfigurationStoreExce
 //				throw new ConfigurationStoreException("Certificate policy group " + group.getPolicyGroupName() + " already exists");
-//			
+//
+//			group.setId(null);
```
is this commented out? Why not just delete?

christophermay07 replied:
Cert policy functionality is currently no-op'ed in our code, but the assumption was that it will eventually be implemented as part of CONNECT; by implementing the CONN-1082 "<entity>.setId(null)" fix here as well (even if just commented out), we don't have to track down the issue again when this class is actually implemented.

msweaver replied:
IMO I would remove this code. We don't have plans to implement right now and our solution will likely need to work for exchange and direct so it's unlikely to follow this same implementation. We can always get the latest code from the RI again.

christophermay07 replied:
Even deleting the commented lines, I assume we need to leave method stubs so that we'll be compliant with Direct's ConfigurationService wsdl.  Is this an acceptable change?

msweaver commented on Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/impl/DNSDaoImpl.java:
```diff
@@ -43,106 +43,103 @@ Neither the name of The Direct Project (directproject.org) nor the names of its
  * @author Greg Meyer
  * @since 1.1
  */
-//@Repository
```
Using this as a guide: http://stackoverflow.com/questions/1440096/difference-between-repository-and-service

is it still correct to remove @Repository? Do we do any manipulation of the objects or perform any service aside from storing/retrieving the data?

christophermay07 replied:
Good catch; it appears both annotations will work in this case, but the Repository annotation is the one intended for use with a DAO.  I'll update the PR with the annotations changes for all DAO Impl classes.

msweaver commented on Product/Production/Adapters/General/CONNECTDirectConfig/src/main/java/gov/hhs/fha/nhinc/directconfig/dao/impl/AnchorDaoImpl.java:
```diff
@@ -178,7 +179,6 @@ public void add(Anchor anchor) {
             log.debug("Calling JPA to persist the Anchor");
 
             sessionFactory.getCurrentSession().persist(anchor);
-            sessionFactory.getCurrentSession().flush();
```
Why do we have to remove this?

christophermay07 replied:
The flush statement is, at best, redundant; a flush happens at commit.  In the code we absorbed, persist was being called on the session without an explicit commit following, hence the flush.  With our switch to JTA-CMT, we let the container handle when to commit and flush; our code's only responsibility is to get the session properly.

Hibernate's documentation offers the following: "With CMT, transaction demarcation is completed in session bean deployment descriptors, not programmatically."
- http://docs.jboss.org/hibernate/core/3.3/reference/en/html/transactions.html#transactions-demarcation-jta

msweaver replied:
ty

christophermay07 committed:
CONN-1082: Annotated DAO classes correctly, added NotImplementedException for no-ops

msweaver commented:
Sorry Chris, two more comments - this will be the best PR ever ;-)
1) Add the CONNECT license
2) Add/update java docs.

christophermay07 committed:
CONN-1082: Added license and missing Override annotations; fixed more formatting issues

christophermay07 commented:
@msweaver Per your comments above:

1) Added the CONNECT license.
2) Every class touched by this PR inherits its interface's javadocs (although I added all missing "Override" annotations to help make this clearer), whether a "non-Javadoc" comment has been added or the "inheritDoc" annotation has been declared explicitly.  For reference:  http://docs.oracle.com/javase/6/docs/technotes/tools/solaris/javadoc.html#inheritingcomments
